### ISSUE TITLE
Create ionian OS for cd/dvd & VM usage

### ISSUE THREAD

**Issue by ckti** [cevap/ion-old#107]
Create an OS based on GItian that can be put on a USB stick/CD/Virtual System a-la other projects.

**Comment by Bige086** [comment]
This will be awesome!

**Comment by cevap** [comment]
yes, that has been already planned and shows how projects and goals are connected. This is essential if we can provide. Until now we already work/published on:

- [X] [PandaOS](https://susestudio.com/appliance/1728916/releases/42.1.3/changelog) (opensuse leap based) (server+desktop gnome/cinnamon/lxde for now)
- [X] IonianOS (latest debian stable based, debian 9 currently) (server+desktop gnome/cinnamon/lxde for now)
- [ ] Archlinux based OS (currently there is no name)
- [ ] ~~Ion Windows OEM (Windows 10)~~ (would require us to have working concept of selling with HW, otherwise it does not make sense and would probably conflict with microsoft's policies)

**Comment by ckti** [comment]
Have started with gitian ubuntu 16.04 LTS  and 16Gb worth of space.

**Comment by ckti** [comment]
have switched over to debian 9 for the base image.

**Comment by ckti** [comment]
debian 9 stretch OS and cinnamon WM.  Getting the base system installed first then starting in on customizations for our purposes.

**Comment by ckti** [comment]
now starting in on the package creation.  will need a place to store these as well.

**Comment by ckti** [comment]
debian package for ioncoin and db4.8 built, syncing with blockchain now

**Comment by ckti** [comment]
Synced with blockchain.  may need to revert back to ubuntu for automatic updates

**Comment by ckti** [comment]
building parallel ubuntu xenial from minimal install. Auto updating ioncoin package (iond &  ion-qt)  from launchpad makes more sense than using a non-official repo.

**Comment by ckti** [comment]
starting customization of xenial image

**Comment by ckti** [comment]
will have to go with an i386 image as well to make the OS compatible with older hardware

**Comment by ckti** [comment]
i386 minimal image created, now to select packages/WM and customize

**Comment by ckti** [comment]
digging into the wonders of plymouth....

**Comment by ckti** [comment]
should have an alpha up earl;y next week.  still digging through plymouth and using gimp to get the correct boot splash screen.

**Comment by ckti** [comment]
boot screen at ~95% on i386

**Comment by ckti** [comment]
will have i386 alpha available today (Ubuntu 16.04 based)

**Comment by ckti** [comment]
ran into an issue copying from vm -> img -> sd ... investigating

**Comment by ckti** [comment]
got it now, i386 version image created and currently syncing with blockchain to ensure 16GB is large enough filesystem.

**Comment by ckti** [comment]
blockchain synced now to re-make images

**Comment by ckti** [comment]
fixed a couple of minor bugs and installing on a usb stick...should be the last iteration of the i386 alpha

**Comment by ckti** [comment]
uploading now to https://ioncoin.000webhostapp.com/Ionian/i386

**Comment by ckti** [comment]
Image to big to host on hostinger have uploaded to https://drive.google.com/file/d/1UXgO9VQRXyMu3IC69VmVHp3VBfTFUyq9/view ... 2gb image alpha system

**Comment by ckti** [comment]
Username ionian password ionian open sudo

**Comment by ckti** [comment]
starting in on amd64 version

**Comment by ckti** [comment]
have efi boot partition working now. found a (very minor) issue in the desktop application where the installation does not show a directory icon.  Will open a new issue for this.

**Comment by ckti** [comment]
had to resize the image to 14GiB this will now fit on any 16GB+ size disk.  Building and testing

**Comment by ckti** [comment]
need to set default background for desktop internally rather than by the user

**Comment by ckti** [comment]
Now to get the boot splash screen to cover the entire boot process (currently drops to text on fsck and goes back to the plymouth boot screen near the end of the boot process {may need to dig into the grub scripting language to do this}).

**Comment by ckti** [comment]
also working on changing lsb_release info

**Comment by ckti** [comment]
Have a new system setup ready.  Copying to USB to test.

**Comment by ckti** [comment]
i386 USB Boot works.  Will post as a beta on google drive today

**Comment by ckti** [comment]
old alpha on google drive has been removed

**Comment by ckti** [comment]
currently xz compressing image

**Comment by ckti** [comment]
Beta Version up at https://drive.google.com/open?id=1Fk44G_mseUDWRC5U1z2Ol2jl_-dYTa72 still username ionian password ionian unlocked sudo some fine tuning needed for the boot graphic and initial desktop panels but working nonetheless.

**Comment by ckti** [comment]
have now got the initial boot image very close and and now testing the default desktop

**Comment by ckti** [comment]
downloading full blockchain to make sure there is enough room on usb stick with 10GB partition

**Comment by ckti** [comment]
Need to clean up the desktop icons

**Comment by ckti** [comment]
desktop icon created. all looks good from that perspective.  now to get the grub boot graphic to match the animated plymouth graphic.

**Comment by ckti** [comment]
New beta will be up today

**Comment by ckti** [comment]
ionian i386 beta availiable at https://drive.google.com/open?id=1Fk44G_mseUDWRC5U1z2Ol2jl_-dYTa72

**Comment by ckti** [comment]
Will use the current i386 version as a guide for x86_64, vdi (32 and 64 bit) and ISO (i386, x86_64) creation.

**Comment by ckti** [comment]
creating i386 vdi

**Comment by ckti** [comment]
i386 ionian VDI is up at https://drive.google.com/open?id=17d7vtFAK9wfbXFjbUEtd6nGr9e33259E

**Comment by ckti** [comment]
Got what I think is a good OS build for amd64.  Transferring it to a smaller hard drive then removing the log files &c, making it a clean OS  then testing on usb stick.

**Comment by ckti** [comment]
have most of what i need now to finalize rpi3 build of ionian

**Comment by ckti** [comment]
Will be removing ionian os from google drive and putting it up on the beta-test web server

**Comment by ckti** [comment]
i386 image and vdi image are now up DM me for site, username and password

**Comment by ckti** [comment]
will have a raspian beta up today.

**Comment by ckti** [comment]
looks like tomorrow before the rpi3 image is up now.

**Comment by ckti** [comment]
solved the issue i was having and am creating the .xz file now.  Will have raspian image up tomight

**Comment by ckti** [comment]
uploading rpi3 .xz image now

**Comment by ckti** [comment]
rpi3 image uploaded. if you want access DM me for site, user and password.

**Comment by ckti** [comment]
have got the boot logo animation ~90% finished.  Minor issue of cutting off outside in a square pattern during final phase of boot.

**Comment by ckti** [comment]
creating new images with new git version

**Comment by ckti** [comment]
Cannot pull down new version of ion until launchpad is back and i can rebuild fully.  Will go with pbuilder locally for the time being

**Comment by ckti** [comment]
now have the plymouth screen 100% will install new ioncoin package starting with i386 .img file

**Comment by ckti** [comment]
need to get the resolution &/or background displaying correctly

**Comment by ckti** [comment]
looking better now, ionian i386 vdi displaying correctly

**Comment by ckti** [comment]
looking better now, ionian i386 vdi displaying correctly

**Comment by ckti** [comment]
usb image looks good as well

**Comment by ckti** [comment]
interesting, added in the packages for full screen plymouth boot (ie from the boot command line) and x-windows stopped working.  reverting to l.k.g. configuration and following the steps to see where it breaks.

**Comment by ckti** [comment]
this explains why this part of ubuntu plymouth is broken.  Have a good config now and will get it working on 16GB USB img

**Comment by ckti** [comment]
2 steps forward 1 step back, apt-get update breaks /etc/default/iond

**Comment by ckti** [comment]
didn't put conffiles in debian

**Comment by ckti** [comment]
..shouldn't make a difference as /etc files are processed as configuration files but am testing to make sure.  At least I'm back to where I was before launchpad went down (even though I'm running pbuilder locally to get it compiled).

**Comment by ckti** [comment]
one-off maybe?  Did an apt-get update will try with prev -> master

**Comment by ckti** [comment]
launchpad cannot find username :(

**Comment by ckti** [comment]
all back online now...only had the one update overwrite /etc/default/iond...will keep an eye on it in the next iteration

**Comment by ckti** [comment]
uploading new vdi now

**Comment by ckti** [comment]
starting back into 16GB disk now

**Comment by ckti** [comment]
Ionian BETA vdi up on ioncore download site, DM for access

**Comment by ckti** [comment]
With the ubuntu plymouth bug I'm having to re-think the animated plymouth boot screen for the usb OS image.

**Comment by ckti** [comment]
Will take a bit of math to figure out the correct path of the animation for the production release, but it will get sorted.  Will have a beta up tomorrow.

**Comment by ckti** [comment]
building beta

**Comment by ckti** [comment]
USB Image is up on website,DM for username password and URL.

**Comment by ckti** [comment]
Got the animation working correctly now. Will have new i386 betas up tomorrow :) The on to the amd64 EFI and DVD isolinux boots

**Comment by ckti** [comment]
Currently compressing i386 images

**Comment by ckti** [comment]
Uploading vdi image now

**Comment by ckti** [comment]
VDI Uploaded, now uploading USB .img file

**Comment by ckti** [comment]
(*&^(*^*(&^ just found a small problem with the initial configuration.  need to remove a unnecessary directory.

**Comment by ckti** [comment]
it's only on the .img not the .vdi

**Comment by ckti** [comment]
.fontconfig directory was still in /home/ioncoin, removed and need to re-compress and re-upload.

**Comment by ckti** [comment]
new version going up now

**Comment by ckti** [comment]
uploaded now, DM me for access....on to amd64 and iso next

**Comment by ckti** [comment]
amd64 needs to be able to dual boot i386 and amd64 working on this now

**Comment by ckti** [comment]
This part (dual boot ia32/amd64) is definitely going to take some time to implement

**Comment by ckti** [comment]
happy with the amd64 os image.

**Comment by ckti** [comment]
building an EFI only amd64 boot disk and vdi

**Comment by ckti** [comment]
should have an alpha amd64 image up this weekend

**Comment by ckti** [comment]
Starting back with a clean install of amd64 on a blank 16GB USB drive.

**Comment by ckti** [comment]
Unetbootin may be a way forward.  Creating disk image of xubuntu xenial now

**Comment by ckti** [comment]
Unetbootin didn;t work....ine run through on a VDI then isolinux

**Comment by ckti** [comment]
have a starting image now.  will start in to the customization process from bare metal install

**Comment by ckti** [comment]
unetbootin is not the way forward, going back the the original amd64 image i have already

**Comment by ckti** [comment]
unetbootin will work for creating dual-boot iso image

**Comment by ckti** [comment]
still getting efi shell & local disk boots on vdi and flash drive.

**Comment by ckti** [comment]
Trying a different approach now

**Comment by ckti** [comment]
i386 boot works for vdi on amd64 OS

**Comment by ckti** [comment]
Uploading amd64 vdi now

**Comment by ckti** [comment]
amd64 vdi is uploaded DM for username/password

**Comment by ckti** [comment]
have amd64 booting with USB now.  Customizing the system and alpha will be up today.

**Comment by ckti** [comment]
still a few more tests to run on the USB image.  Will have an alpha version available tomorrow

**Comment by ckti** [comment]
uploading usb image now

**Comment by ckti** [comment]
uploaded alpha version of USB ionian DM for access.

**Comment by ckti** [comment]
Creating archives of the customizations to start in to a production i386 build

**Comment by ckti** [comment]
customizations done, creating vdi and usb images

**Comment by ckti** [comment]
autologin now in place rather than having to type startx at the cli.

**Comment by ckti** [comment]
creating usb image to fit 16G+ usb drives

**Comment by ckti** [comment]
Have put in a gcc-7 build restriction in xenial packages.  Waiting for libdb to sync then will upload version 4

**Comment by ckti** [comment]
Uploading stable release versions of Ionian i386 amd64 to follow

**Comment by ckti** [comment]
i386 Ionian release uploaded...DM for url username and password

**Comment by ckti** [comment]
uploading amd64 vdi zip file

**Comment by ckti** [comment]
creating amd64 USB

**Comment by ckti** [comment]
amd64 release is now uploaded DM for url, username and password.

**Comment by ckti** [comment]
compressing usb image for upload

**Comment by ckti** [comment]
uploading usb image now

**Comment by ckti** [comment]
usb image uploaded DM for user, site, and password

**Comment by ckti** [comment]
Starting into the iso images one for i386 and one for amd64

**Comment by ckti** [comment]
Will have the ISO images install Ionian to the hard drive rather than have a live-mount image (there are two live mounts for each of i386 and amd64 already).  This part will take a while to do.

**Comment by ckti** [comment]
Looks like I can do a live image with an install option a-la other Ubuntu distros.

**Comment by ckti** [comment]
Have a starting image to use.  Customization is going to take some time

**Comment by ckti** [comment]
changing one thing at a time, creating the ISO and booting into it.

**Comment by ckti** [comment]
yes, these iso images will take a while to get out

**Comment by ckti** [comment]
may have found a new way to create these isos...watch this space

**Comment by ckti** [comment]
Need to start back over again.....

**Comment by ckti** [comment]
About 85% of the way through now.

**Comment by ckti** [comment]
One small piece to sort out --- the initrd.lz file is being a pain though.  WIll upload a beta i386 ISO image either tonight or tomorrow

**Comment by ckti** [comment]
I have to change the install panels as well.  This will take a while as well.

**Comment by ckti** [comment]
Grub screen when hitting tab needs to be changed too.

**Comment by ckti** [comment]
Looks like next week before I get a clean alpha i386 ISO up

**Comment by ckti** [comment]
Have a good boot image using the existing ionian OS on img/vdi...need to debug ubiquity to get the install from live cd working.

**Comment by ckti** [comment]
ubuquity doesn't like my custom CD alterations.

**Comment by ckti** [comment]
got it installing now, having crashes due to removing the ubiquity slide show package. resinstalling and rebuilding.  once it installs properly i can move to creating dvds and installing on bare-metal systems

**Comment by ckti** [comment]
got the i386 ISO booting and installing from DVD-rw now using stock ubuntu ubiquity slide show package.  Will put up and alpha of this today as long as nothing else pops up during my testing.

**Comment by ckti** [comment]
Uploading alpha ISO now

**Comment by ckti** [comment]
Alpha i386 ISO uploaded, DM for username, password & Site

**Comment by ckti** [comment]
Working on amd64 ISO image now

**Comment by ckti** [comment]
uploading ALPHA i386 & amd64 images

**Comment by ckti** [comment]
Images are uploaded, DM for username passsword and URL

**Comment by ckti** [comment]
Couple of minor issues to address with amd64 EFI dual boot ISO, boot is very slow to start X-Windows dual-boot not 100% on every system yet.  Ran into an issue with CSM/UEFI system bios.

**Comment by ckti** [comment]
got it dual booting EFI/el-torito some customizations are now needed to finalize the ISO build on both i386 and amd64.  Can then re-visit usb images and finalize their customizations.

**Comment by ckti** [comment]
Will have a new amd64 BETA ISO image up today

**Comment by ckti** [comment]
ISO Image being uploaded now

**Comment by ckti** [comment]
Beta amd64 iso is now uploaded DM for username password and site

**Comment by ckti** [comment]
now working on the i386 ISO

**Comment by ckti** [comment]
Uploading i386 ISO

**Comment by ckti** [comment]
i386 ISO uploaded and ready to go.  DM for username/password & url

**Comment by ckti** [comment]
still some minor customizations to do in usb/vdi and ISOs

**Comment by ckti** [comment]
vdi images are finished and am uploading now

**Comment by ckti** [comment]
Uploads didn't go through as they should have, re-uploading now

**Comment by ckti** [comment]
Finally, uploads are working, uploading amd64 and i386 vdi images

**Comment by ckti** [comment]
vdi images now uploaded (finally)

**Comment by ckti** [comment]
still working on the USB boot images

**Comment by ckti** [comment]
very close to a final amd64 usb image.  Will be useable on anything from an 8GB drive upwards.  Will have to make the same changes and update the i386 usb once amd64 is finished

**Comment by ckti** [comment]
looks like i'm going to have to build an iso hybrid with caper-rw for amd64

**Comment by ckti** [comment]
two steps forward and one back, will go with the usb.img root filesystem that already exists make a few more configuration tweaks and see which works better, usb .img or isohybrid/casper-rw when booting system it just takes 1/2 hour each time to write and test the usb filesystem image even on usb3 and it takes approximately the same amount of time to build and test using squashfs on the isohybrid system

**Comment by ckti** [comment]
there's always elilo as well....

**Comment by ckti** [comment]
Going to get the i386 image updated and uploaded today whilst I tear my hair out over amd64 EFI

**Comment by ckti** [comment]
Putting new amd64 ISO up on website

**Comment by ckti** [comment]
New i386 USB is uploaded and ready to go

**Comment by ckti** [comment]
Have found the way forward, usb+grub2+efi for amd64 usb image

**Comment by ckti** [comment]
one more test to go and then i will compress and upload the amd64 image.

**Comment by ckti** [comment]
image will take some time to compress and upload

**Comment by ckti** [comment]
Uploading amd64 image now.

**Comment by ckti** [comment]
All images created and uploaded.  
Ionian-i386-01.03.18.usb.img.xz
Ionian.amd64-02.03.18.img.xz
Ionian-16.0.4-3.1e.amd64.iso
ionian-16.0.4.3-2e.i386.iso
Ionian-amd64-21.02.18.vdi.zip
Ionian-i386-21.02.18.vdi.zip
ionian-2.1.6.3-2655d30.rpi3.img.xz

**Comment by ckti** [comment]
Have opened issue number https://github.com/cevap/ion/issues/153 to deal with the updating of these images
